internal/services/availability_status_service.go

```diff
@@ -1,29 +1,50 @@
 package services
 
 import (
+	"errors"
 	"net/http"
 
 	"github.com/RHEnVision/provisioning-backend/internal/background"
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
 	"github.com/RHEnVision/provisioning-backend/internal/kafka"
-	"github.com/RHEnVision/provisioning-backend/internal/metrics"
 	"github.com/RHEnVision/provisioning-backend/internal/payloads"
 	"github.com/go-chi/render"
 )
 
 func AvailabilityStatus(w http.ResponseWriter, r *http.Request) {
+	logger := ctxval.Logger(r.Context())
 	payload := &payloads.AvailabilityStatusRequest{}
 	if err := render.Bind(r, payload); err != nil {
 		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), "availability status", err))
 		return
 	}
 
+	// Get sources client
+	sourcesClient, err := clients.GetSourcesClient(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewClientError(r.Context(), err))
+		return
+	}
+
+	// Fetch authentication from Sources
+	_, err = sourcesClient.GetAuthentication(r.Context(), payload.SourceID)
+	if err != nil {
+		if errors.Is(err, clients.NotFoundErr) {
+			logger.Warn().Msg("Irrelevant request from sources has received")
+			renderError(w, r, payloads.NewNotFoundError(r.Context(), "Not found error", err))
+			return
+		}
+		renderError(w, r, payloads.NewClientError(r.Context(), err))
+		return
+	}
+
```

Comment:
I would *really* like to keep this service ultra-fast just enqueing messages into Kafka, can we do this in the background job?

Reply:
I liked this approach as we could use the HTTP status code to filter out the incoming requests, but if we add a metric in the background job, we are fine I guess :)

Reply:
I am fine if you want it this way, it is okay too. It is just it can keep our REST workers more busy than necessary but it's fine.

Reply:
I moved the check to the background job, I wanted to move the `IsProvisioningSource` function to be a part of the sources interface but a cyclic import is present that way. 
Do you have any preferences or suggestions? |
I divided the change into four commits, one of them is the relevant one